### ISSUE TITLE
Unable to update note 

### ISSUE THREAD

**Issue by sleekybadger** [NARKOZ/gitlab#304]
Hi. I tried to update merge request comment via notes api and looks like current behavior not match [documentation](https://github.com/NARKOZ/gitlab/blob/master/lib/gitlab/client/notes.rb#L256):

```ruby
client = Gitlab.client(endpoint: 'https://gitlab.com/api/v4', private_token: 'token')
client.edit_merge_request_comment(project_id, merge_request_id, note_id, 'New body')

Gitlab::Error::BadRequest: Server responded with code 400, message: body is missing. Request URI: https://gitlab.com/api/v4/projects/project_id/merge_requests/merge_request_id/notes/note_id
```

However, I can update comment if i will pass body in hash: 

```ruby
client = Gitlab.client(endpoint: 'https://gitlab.com/api/v4', private_token: 'token')
client.edit_merge_request_comment(project_id, merge_request_id, note_id, { body: 'New body' })
```

This behavior same for other note update methods, but not for [create methods](https://github.com/NARKOZ/gitlab/blob/master/lib/gitlab/client/notes.rb#L121). So I wonder, should we change code to match documentation and other notes methods or change documentation?

**Comment by asedge** [comment]
@sleekybadger If `body` is required when editing/updating a note, we should update the code so it follows the pattern in the create methods.  If it's optional we should update the documentation.  Does that make sense?  Will you be putting in a PR to fix this?

**Comment by sleekybadger** [comment]
@asedge Yep, it's required. Sure, it's [here](https://github.com/NARKOZ/gitlab/pull/305)

**Comment by asedge** [comment]
Oh crap.  I made a mistake.  I didn't even look at the behavior defined in the tests.  We cannot break the API as this will be a minor release.  You could make it so that we support both a String or a Hash with a `:body` key, but that could get a bit messy.  The other option is to just go back and update the documentation.  Sorry about this!

**Comment by sleekybadger** [comment]
@asedge sure, pr updated

**Comment by asedge** [comment]
@sleekybadger Sorry for letting this linger for so long.  I have merged #354 which should allow us to migrate to a consistent API while still warning users that passing the note content in a `Hash` is deprecated.  Thanks for bringing this to my attention and your attempts at fixing it - despite my lack of clear decision making!
